### PR TITLE
RDoc-699 Security skeleton + overview draft

### PR DESCRIPTION
**Pull request by iftahbe**

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+[Authentication in the Cluster](authentication/authentication-in-the-cluster)
+
+[Common Errors & Troubleshooting](authentication/common-errors-and-troubleshooting)
+
+
+## Authorization
+
+RavenDB uses x.509 certificates for authorization as well.
+
+It is considered more secured to use client certificates for authorization, insdead of having usernames and passwords.
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown]: `insdead ` -> instead

Comment (ayende) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown]: In general, you can drop this entire sentence

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authorization/security-clearance-and-permissions.markdown [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+The certificate is used for authentication, validating that users are who they say they are. Once a request is authenticated, RavenDB determines the security clearance and permissions for that certificate and allows access according to permissions the admin assigned when setting up the client certificate. 
+
+A client certificate's security clearance can be one of the following:
+
+"Cluster Admin"
+"Operator"
+"User"
+
+*The server certificate is called "Cluster Node" in the UI.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authorization/security-clearance-and-permissions.markdown]: This does not sound right. Is that true?

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authorization/security-clearance-and-permissions.markdown [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+An "Operator" can also manage all the databases, but is unable to modify the cluster itself. It cannot perform operations such as add/remove/promote/demote nodes from the cluster, using the JS admin console, and managing certificates with "Cluster Admin" clearance. 
+
+A certificate with "Operator" clearance is typically used in a hosted solution. If you are running on your own machines, you'll typically ignore that level in favor of "Cluster Admin" or "User"
+
+A certificate with a "User" security clearance can only access specific databases, and only with specific permissions assigned to that certificate. It cannot perform any admin operations at the cluster level. 
+
+
+The type of permissions given to a "User" certificate are "Database Admin" or "read/write". Different permissions are given to a "User" for different databases. A "User" cannot access databases which are not defined in its permissions. The one who sets these permissions is the one that generates the "User" client certificate (either "Cluster Admin" or "Operator").
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authorization/security-clearance-and-permissions.markdown]: Different permissions are given to a "User" for different databases. => Permissions can be assigned to a "User" per database basis.

?

File: Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# Overview
+
+In the security section, we will review the security features in RavenDB and explain how to manage your secured server or cluster.
+
+## Authentication
+
+RavenDB uses x.509 certificate-based authentication.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown]: typo: x.509 => X.509

File: Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown [hunk 5]
```diff
@@ -0,0 +1,63 @@
+# Overview
+
+In the security section, we will review the security features in RavenDB and explain how to manage your secured server or cluster.
+
+## Authentication
+
+RavenDB uses x.509 certificate-based authentication. 
+X.509 certificates are standardized, secured and widely used in many applications. They allow you to use TLS/SSL and HTTPS which keeps your communications encrypted and secured.
+
+The idea of authentication in RavenDB is that the server holds the "server certificate" which was signed by a trusted SSL Certificate Authority or it can be self-signed.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown]: I think we need to describe more the 'idea'. Now it looks like the whole idea is that server has a certificate.

Maybe merge this with next paragraph?

E.g.
The idea of authentication in RavenDB is based on a fact that server holds a certificate (known as "server certificate"), signed by a trusted SSL Certificate Authority or self-signed. The server certificate is used by an administrator to generate client certificates with assigned permissions and access. The client certificate can be used  for authentication and authorization is granted according to the assigned permissions and access level (read more about Authorization in [next section]()).

File: Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+[Client Certificate Usage](authentication/client-certificate-usage)
+
+[Authentication in the Cluster](authentication/authentication-in-the-cluster)
+
+[Common Errors & Troubleshooting](authentication/common-errors-and-troubleshooting)
+
+
+## Authorization
+
+RavenDB uses x.509 certificates for authorization as well.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown]: typo: x.509 => X.509

File: Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+[Security Clearance & Permissions](authorization/security-clearance-and-permissions)
+
+[List of Security Clearances](authorization/list-of-security-clearances)
+
+
+## Encryption
+
+RavenDB offers full database encryption using libsodium. 
+
+Encryption is implemented at the storage level, with the ChaCha20Poly1305 authenticated encryption scheme and 256 bits keys.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown]: authenticated encryption scheme?
typo: 256 bits => 256 bit

File: Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+#### Read more:
+
+[Security Clearance & Permissions](authorization/security-clearance-and-permissions)
+
+[List of Security Clearances](authorization/list-of-security-clearances)
+
+
+## Encryption
+
+RavenDB offers full database encryption using libsodium.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown]: RavenDB offers full database encryption using libsodium. 
=>
RavenDB offers full database encryption using [libsodium](), a well-known battle tested encryption library.

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authorization/security-clearance-and-permissions.markdown (hunk 2 above)
Comment (iftahbe) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/authorization/security-clearance-and-permissions.markdown]: It's true

File: Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown (hunk 5 above)
Comment (iftahbe) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/overview.markdown]: Yeah that's good, I'll do that

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authorization/security-clearance-and-permissions.markdown (hunk 2 above)
Comment (ppekrol) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/authorization/security-clearance-and-permissions.markdown]: Just saying that for a person that is not familiar with certificates (me) it does not sound correct.

Comment (iftahbe) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/authorization/security-clearance-and-permissions.markdown]: It's not the server certificate that is called that but the security clearance for that certificate. I will rephrase

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            "Password": "s3cr7t p@$$w0rd"
+        }
+    }  
+
+
+The second way to enable authentication is to set `Security.Certificate.Exec`. 
+
+This option is useful when you want to protect your certificate (private key) with other solutions such as "Azure Key Vault", "HashiCorp Vault" or even Hardware-Based Protection. RavenDB will invoke a process you specify, so you can write your own scripts / mini programs and apply whatever logic you need. It creates a clean separation between RavenDB and the secret store in use.
+
+RavenDB excpects to get the raw binary representation (byte array) of the .pfx certificate through the standard output.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown]: typo: excpects -> expects

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+Let's look at an example - a simple powershell script called give_me_cert.ps1
+
+    try
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown]: Would be better to have it in CODE-BLOCK

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+{NOTE This section explains how to setup authentication <strong>manually</strong>. Please also take a look at the automated [Setup Wizard](../../../start/setup-wizard) which lets you setup authentication in a much easier and faster way. /}
+
+To enable authentication, either `Security.Certificate.Path` or `Security.Certificate.Exec` must be set in settings.json.
+
+The first way to enable authentication is to set `Security.Certificate.Path` with the path to your .pfx server certificate. You may supply the certificate password using `Security.Certificate.Password`. 
+When providing a certificate for authentication, you <strong>must</strong> also set the `ServerUrl` configuration option to an HTTPS address.
+
+For example, this is a typical settings.json:
+
+    {
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown]: Would be better to have it in CODE-BLOCK... IIRC {CODE-BLOCK:json}...{CODE-BLOCK/}

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    catch
+    {
+	write-error $_.Exception
+	exit 3
+    }
+
+And settings.json will look something like this:
+
+    {
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown]: {CODE-BLOCK}

**Pull request merged**